polar_route/vessel_performance/vessel_performance_modeller.py

```diff
@@ -76,4 +81,45 @@ def filter_nans(self):
         for i, cellbox in enumerate(self.env_mesh.agg_cellboxes):
             if any(np.isnan(val) for val in cellbox.agg_data.values() if type(val) == float):
                 filtered_data = {k: 0. if np.isnan(v) else v for k, v in cellbox.agg_data.items() if type(v) == float}
-                self.env_mesh.update_cellbox(i, filtered_data)
\ No newline at end of file
+                self.env_mesh.update_cellbox(i, filtered_data)
+
+    def split_neighbouring_cells(self, inaccessible_nodes):
+        """
+        Method to split any accessible cells that neighbour inaccessible cells until their sizes match
+
+        Args:
+            inaccessible_nodes (list): List of inaccessible nodes to split around
+
+        """
+        for in_node in inaccessible_nodes:
+            in_cb = self.env_mesh.get_cellbox(in_node)
+            neighbour_nodes = self.get_all_neighbours(in_node)
+            neighbours = [self.env_mesh.get_cellbox(nn) for nn in neighbour_nodes]
+            # Only interested in splitting accessible neighbours
+            acc_neighbours = [n for n in neighbours if not n.agg_data['inaccessible']]
+            # Assume aspect ratios match and compare widths
+            while any(neighbour.boundary.get_width() > in_cb.boundary.get_width() for neighbour in acc_neighbours):
```

Comment:
Could we have this more generalisable for the get_width and get_height. I think in our splitting at the moment it is ok, but may create an issues later :)